flytekit/contrib/notebook.py

```diff
@@ -0,0 +1,558 @@
+                    cpu_request
+                )
+            )
+        if gpu_request:
+            requests.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.GPU,
+                    gpu_request
+                )
+            )
+        if memory_request:
+            requests.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.MEMORY,
+                    memory_request
+                )
+            )
+
+        limits = []
+        if storage_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.STORAGE,
+                    storage_limit
+                )
+            )
+        if cpu_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.CPU,
+                    cpu_limit
+                )
+            )
+        if gpu_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.GPU,
+                    gpu_limit
+                )
+            )
+        if memory_limit:
+            limits.append(
+                _task_models.Resources.ResourceEntry(
+                    _task_models.Resources.ResourceName.MEMORY,
+                    memory_limit
+                )
+            )
+
+        return _sdk_runnable.SdkRunnableContainer(
+            command=[],
+            args=[],
+            resources=_task_models.Resources(limits=limits, requests=requests),
+            env=environment,
+            config={}
+        )
+
+
+def spark_notebook(
+        notebook_path,
+        inputs={},
+        outputs={},
+        spark_conf=None,
+        cache_version='',
+        retries=0,
+        deprecated='',
+        cache=False,
+        timeout=None,
+        environment=None,
+):
+    """
+    Decorator to create a Notebook spark task. This task will connect to a Spark cluster, configure the environment,
+    and then execute the code within the notebook_path as the Spark driver program.
+    """
+    return SdkNotebookSparkTask(
+            notebook_path=notebook_path,
+            inputs=inputs,
+            outputs=outputs,
+            spark_conf=spark_conf,
+            discovery_version=cache_version,
+            retries=retries,
+            deprecated=deprecated,
+            discoverable=cache,
+            timeout=timeout or _datetime.timedelta(seconds=0),
+            environment=environment or {},
+        )
+
+
+def _find_instance_module():
+    frame = _inspect.currentframe()
+    while frame:
+        if frame.f_code.co_name == '<module>':
+            return frame.f_globals['__name__']
+        frame = frame.f_back
+    return None
+
+
+class SdkNotebookSparkTask(SdkNotebookTask):
```

Comment:
Can you explain what the fundamental difference is between a spark notebook and a non-spark notebook?  If I run a regular notebook but my workflow image has all the spark functionality, can it transform interactively into a spark notebook?

Reply:
Agreed. I'd strongly recommend following through on the already open PR which elevates notebooks to provide execution logic for all tasks.

Reply:
@matthewphsmith do you think you have the time to merge your PR then? We want to take it, but no one has any context. If you can take it, test, respond to comments and merge we would love to take that